finagle-netty3/src/main/scala/com/twitter/finagle/netty3/ssl/client/SslClientConnectHandler.scala

```diff
@@ -113,26 +112,27 @@ private[netty3] class SslClientConnectHandler(
                 )
               )
             } else {
-              try {
-                if (sessionVerifier(address, config, engine.getSession)) {
-                  connectFuture.get.setSuccess()
-                  SslClientConnectHandler.super.channelConnected(ctx, e)
-                } else {
-                  fail(
-                    ctx.getChannel,
-                    new SslVerificationFailedException(
-                      new Exception("Failed client verification"),
-                      ctx.getChannel.getRemoteAddress
+              sessionVerifier(address, config, engine.getSession)
+                .onSuccess { verifierResult =>
```

Comment:
We generally use `Future.respond` in code within Finagle as opposed to `Future.onSuccess` and `Future.onFailure`, especially when there is code for both branches. Would you mind switching this review to use `Future.respond` instead?